### PR TITLE
add Add to Collection plugin

### PR DESCRIPTION
**Pull request by dvirtz**
This plugin adds any saved release to one of the user collections

### CODE REVIEW

File: plugins/add_to_collection/override_module.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+from typing import Generator
+
+
+@contextmanager
+def override_module(obj: object) -> Generator[None, None, None]:
+    # picard expects hooks to be defined at module level
+    module = obj.__module__
+    obj.__module__ = ".".join(module.split(".")[:-1])
+    yield
+    obj.__module__ = module
```
Comment (phw) [inline plugins/add_to_collection/override_module.py:11]: I guess it's fine and I understand why you wanted to split this appart as you did. But it also feels a bit like over engineering for this plugin, to be honest.

Anyway, I'm fine keeping this.

**Review by phw — APPROVED** [review]
Thanks a lot for this. This is definitely useful and it is looking good. LGTM

Overall this is some functionality that eventually should end up in Picard at some point. But at this moment I think having it as a plugin and letting that mature first is a good move.

For a future idea I think it would be cool if one could define some conditions to have saved releases end up in different collections based on some criteria. E.g. I keep my physical releases in a separate collection from the downloads.

**Review by zas — APPROVED** [review]
LGTM

Clearly this is something Picard should handle natively, including adding one release to multiple collections.

File: plugins/add_to_collection/override_module.py (hunk 1 above)
Comment (dvirtz) [reply plugins/add_to_collection/override_module.py:11]: I hope this limitation can be lift off in [V3](https://github.com/phw/picard/blob/plugins-v3/PLUGINS.md)

Comment (phw) [reply plugins/add_to_collection/override_module.py:11]: Yes, I will look into this when I get back to the reworked plugin handling

**Pull request merged**